dot/state/service.go

```diff
@@ -149,10 +164,43 @@ func (s *Service) Initialize(data *genesis.Data, header *types.Header, t *trie.T
 	return nil
 }
 
+func (s *Service) loadBabeConfigurationFromRuntime(t *trie.Trie, db chaindb.Database, gen *genesis.Genesis) (*types.BabeConfiguration, error) {
+	// load genesis state into database
+	genTrie, err := rtstorage.NewTrieState(chaindb.NewTable(db, storagePrefix), t)
+	if err != nil {
+		return nil, fmt.Errorf("failed to instantiate TrieState: %w", err)
+	}
+
+	// create genesis runtime
+	rtCfg := &wasmer.Config{}
+	rtCfg.Storage = genTrie
+	rtCfg.LogLvl = 4
+
+	r, err := wasmer.NewRuntimeFromGenesis(gen, rtCfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create genesis runtime: %w", err)
+	}
+
+	// load and store initial BABE epoch configuration
+	babeCfg, err := r.BabeConfiguration()
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch genesis babe configuration: %w", err)
+	}
+
+	r.Stop()
+
```

Comment:
What is r.Stop() stopping?  I don't see when/where r was started?

Reply:
`r` is the runtime instance, it's declared a few lines up: `r, err := wasmer.NewRuntimeFromGenesis(gen, rtCfg)`